pkg/infrastructure/services/resources.go

```diff
@@ -0,0 +1,183 @@
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"reflect"
+)
+
+const (
+	enablePersistenceEnvKey = "ENABLE_PERSISTENCE"
+)
+
+// createRequiredResources creates the required resources given the KogitoService instance
+func (s *serviceDeployer) createRequiredResources(instance v1alpha1.KogitoService) (resources map[reflect.Type][]resource.KubernetesResource, err error) {
+	resources = make(map[reflect.Type][]resource.KubernetesResource)
+	imageHandler := newImageHandler(instance, s.definition.DefaultImageName, s.client)
+	deployment := createRequiredDeployment(instance, imageHandler)
+	if s.definition.CustomConfigOnDeploymentHook != nil {
+		if err = s.definition.CustomConfigOnDeploymentHook(deployment, instance); err != nil {
+			return
+		}
+	}
+
+	service := createRequiredService(instance, deployment)
+
+	if s.definition.infinispanIntegration {
+		if err = s.applyInfinispanConfigurations(deployment, instance); err != nil {
+			return
+		}
+	}
+
+	if s.definition.kafkaIntegration {
+		if instance.GetSpec().(v1alpha1.KafkaAware).GetKafkaProperties().UseKogitoInfra &&
+			infrastructure.IsStrimziAvailable(s.client) {
+			if topics, err := s.createKafkaTopics(instance); err != nil {
+				return nil, err
+			} else if topics != nil {
+				resources[reflect.TypeOf(kafkabetav1.KafkaTopic{})] = topics
+			}
+		}
+		if err = s.applyKafkaConfigurations(deployment, instance); err != nil {
+			return
+		}
+	}
+
+	resources[reflect.TypeOf(appsv1.Deployment{})] = []resource.KubernetesResource{deployment}
+	resources[reflect.TypeOf(corev1.Service{})] = []resource.KubernetesResource{service}
+
+	if imageHandler.hasImageStream() {
+		resources[reflect.TypeOf(imgv1.ImageStream{})] = []resource.KubernetesResource{imageHandler.imageStream}
+	}
+	if s.client.IsOpenshift() {
+		resources[reflect.TypeOf(routev1.Route{})] = []resource.KubernetesResource{createRequiredRoute(instance, service)}
+	}
+
+	return
+}
+
+func (s *serviceDeployer) applyInfinispanConfigurations(deployment *appsv1.Deployment, instance v1alpha1.KogitoService) error {
+	var infinispanSecret *corev1.Secret
+	infinispanAware := instance.GetSpec().(v1alpha1.InfinispanAware)
+	infinispanSecret, err := infrastructure.FetchInfinispanCredentials(infinispanAware, instance.GetNamespace(), s.client)
+	if err != nil {
+		return err
+	}
+	infrastructure.SetInfinispanVariables(
+		infinispanAware.GetInfinispanProperties(),
+		infinispanSecret,
+		&deployment.Spec.Template.Spec.Containers[0])
+
+	if infinispanAware.GetInfinispanProperties().UseKogitoInfra || len(infinispanAware.GetInfinispanProperties().URI) > 0 {
+		framework.SetEnvVar(enablePersistenceEnvKey, "true", &deployment.Spec.Template.Spec.Containers[0])
+	}
+	return nil
+}
+
+func (s *serviceDeployer) applyKafkaConfigurations(deployment *appsv1.Deployment, instance v1alpha1.KogitoService) error {
+	URI, err := getKafkaServerURI(instance.GetSpec().(v1alpha1.KafkaAware).GetKafkaProperties(), s.definition.Namespace, s.client)
```

Comment:
We search for the Kafka Service owned by the Kafka CR, owned by the Kogito Operator if the service is relying on the Infra, otherwise we look for the external URI defined by the user in the CR.